app/mainAppWindow/index.js

```diff
@@ -323,12 +328,16 @@ function setImgSrcSecurityPolicy(policies) {
  * @param {Electron.BeforeSendResponse} callback
  */
 function onBeforeSendHeadersHandler(detail, callback) {
-	if (detail.url.startsWith(customBGServiceUrl.href)) {
-		detail.requestHeaders['Access-Control-Allow-Origin'] = '*';
+	if (intune.isSsoUrl(detail.url)) {
+		intune.addSsoCookie(logger, detail, callback);
+	} else {
+		if (detail.url.startsWith(customBGServiceUrl.href)) {
+			detail.requestHeaders['Access-Control-Allow-Origin'] = '*';
+		}
+		callback({
```

Comment:
I think this callback was happening before even if the case of the not been an if, is this change intended?

Reply:
In the original implementation the callback call happened always. The condition only added a request header, After the changes the old part is left intact, but the SSO part is calling the callback internally. This is needed because the DBus call code is asynchronous, which means that the addSsoCookie method returns immediatelly without calling the callback, but the call is deferred to when the DBus reply (or error) is received.

Reply:
Maybe it would make sense to throw a comment into line 332 explaining that, so that it is clear for the next dev, that will be working with this function.

Reply:
left a comment in there but not a big deal so resolving.